src/plugins/auto/plugin_config.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "utils/plugin_config.hpp"
+
+namespace MultiDevicePlugin {
+const std::set<std::string> PluginConfig::_availableDevices = {"AUTO", "CPU", "GPU", "TEMPLATE", "MYRIAD", "VPUX", "MULTI", "HETERO", "mock"};
+
+PluginConfig::PluginConfig() {
+    set_default();
+    device_property_validator = std::dynamic_pointer_cast<BaseValidator>(std::make_shared<FuncValidator>([](const ov::Any& target) -> bool {
+        auto deviceName = target.as<std::string>();
+        return _availableDevices.end() != std::find(_availableDevices.begin(), _availableDevices.end(),
+                                                            DeviceIDParser(deviceName).getDeviceName());
+    }));
+}
+
+void PluginConfig::set_default() {
+    register_property(
+        std::make_tuple(ov::enable_profiling, false),
+        std::make_tuple(ov::device::priorities, ""),
+        std::make_tuple(ov::hint::model_priority, ov::hint::Priority::MEDIUM),
+        std::make_tuple(ov::log::level, ov::log::Level::NO),
+        std::make_tuple(ov::intel_auto::device_bind_buffer, false),
+        std::make_tuple(ov::hint::performance_mode, ov::hint::PerformanceMode::UNDEFINED, PerformanceModeValidator()),
+        std::make_tuple(ov::hint::num_requests, 0),
+        std::make_tuple(ov::intel_auto::enable_startup_fallback, false),
+        // TODO 1) cache_dir 2) allow_auto_batch 3) auto_batch_timeout
+        std::make_tuple(ov::cache_dir, ""),
+        std::make_tuple(ov::hint::allow_auto_batching, true),
+        std::make_tuple(ov::auto_batch_timeout, 1000),
+        // Legacy API properties
+        std::make_tuple(exclusive_asyc_requests, false));
+}
+void PluginConfig::register_property_impl(const ov::AnyMap::value_type& property, BaseValidator::Ptr validator) {
+    property_validators[property.first] = validator;
+    internal_properties[property.first] = property.second;
+}
+
+void PluginConfig::set_property(const ov::AnyMap& properties) {
+    for (auto& kv : properties) {
+        auto& name = kv.first;
+        auto& val = kv.second;
+        if (is_supported(kv.first)) {
+            OPENVINO_ASSERT(property_validators.at(name)->is_valid(val),
+                    "[AUTO]", "Invalid value for property ", name,  ": ", val.as<std::string>());
+            internal_properties[name] = val;
+            // when user call set_property to set some config to plugin, we also respect this and pass through the config in this case
+            user_properties[name] = val;
+        } else {
```

Comment:
take key perf_count for example
case1: core.set_property(auto, perf_count)
          when auto compile model to CPU, auto should pass the above property through
case 2, compile_model(auto, perf_count)
           later when auto compile model to CPU, auto should pass the above property through
case 3, not set property to auto and no perf count config set with compiled model with auto
           later when auto compile model to CPU, auto should **NOT** pass the above property through
is my understanding corrent? @yangwang201911  @wangleis

Reply:
@songbell perf_count is HW independent property. So in case 3, AUTO will pass default perf_count in AUTO to HW plugin. Your understanding for case 3 is right for HW related property.